### ISSUE TITLE
Fix PostFragment to PublishPostIntentService problem

### ISSUE THREAD

**Issue by reustonium** [hhroc/yellr-android#43]
Reproduce by starting app, clicking on the 'New Post' action button in the HomeActivity.  Then add text to the 'What's on your mind' edittext.  Press upload.

The upload button fires PostFragment.java at line 116.  The code dies in PublishPostIntentService during mediaObject upload at line 245 citing an empty text field.

**Comment by reustonium** [comment]
didn't see anything on this one from last night, so I'm gonna take a crack at it unless you object @thequbit

**Comment by thequbit** [comment]
Why don't I take this one, since it's server communications and I'll have more granularity on what isn't getting t the server.

Reassigning to me @reustonium

**Comment by thequbit** [comment]
So this was 100% my fault.  The docs aren't very good around these two end points.

If `media_type` is "text" then `media_file` and `media_caption` both become optional (they are not looked at by the server).

if `media_type` is not "text" (that is "audio", "video", or "image"), then `media_text` becomes optional (it is not looked at by the server).

It appears that the MultipartEntityBuilder,addPart() function does not accept an empty value when setting parameters (which is different than what Google Chrome's PostMan, where empty fields are allowed).  I wrapped the fields in if statements, and it will be our job to ensure that we just include the correct fields with the correct `media_type`.  I also added default values of `<empty_string>` to the MediaObjectDefinition class fields.

Again, sorry about that Pal, but it is working now.  Pull the code from me and and confirm correct operation, and close at your convenience.
